dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Machines/VoltageRegulators/Standard/ExcIEEEST4B.mo

```diff
@@ -0,0 +1,172 @@
+  Modelica.Blocks.Math.Min min1 annotation(
+    Placement(visible = true, transformation(origin = {150, -60}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.Blocks.Math.Product product1 annotation(
+    Placement(visible = true, transformation(origin = {90, -80}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.Blocks.Math.Product product2 annotation(
+    Placement(visible = true, transformation(origin = {-90, -120}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Dynawo.NonElectrical.Blocks.NonLinear.RectifierRegulationCharacteric rectifierRegulationCharacteric "FEX = F(IN)" annotation(
+    Placement(visible = true, transformation(origin = {30, -120}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.ComplexBlocks.ComplexMath.RealToComplex realToComplex annotation(
+    Placement(visible = true, transformation(origin = {-150, -80}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Dynawo.NonElectrical.Blocks.Continuous.PIAntiWindup2 limPI1(Ki = Kir, Kp = Kpr, Y0 =  Kg *Efd0Pu, uMax = VrMaxPu, uMin = VrMinPu)  annotation(
+    Placement(visible = true, transformation(origin = {-90, 0}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.Blocks.Continuous.FirstOrder firstOrder(T = tR, y_start = Us0Pu) annotation(
+    Placement(visible = true, transformation(origin = {-210, 0}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.Blocks.Math.Min min annotation(
+    Placement(visible = true, transformation(origin = {150, 0}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.Blocks.Sources.Constant constant2(k = UOel0Pu) annotation(
+    Placement(visible = true, transformation(origin = {90, 50}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.Blocks.Continuous.FirstOrder firstOrder1(T = tA, y_start =  Kg *Efd0Pu) annotation(
+    Placement(visible = true, transformation(origin = {-30, 0}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.Blocks.Math.Product product annotation(
+    Placement(visible = true, transformation(origin = {210, 0}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  NonElectrical.Blocks.Continuous.PIAntiWindup2 limPI2(Ki = Kim, Kp = Kpm, Y0 = Efd0Pu / Ub0Pu, uMax = VmMaxPu, uMin = VmMinPu) annotation(
+    Placement(visible = true, transformation(origin = {90, 0}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.Blocks.Math.Feedback feedback annotation(
+    Placement(visible = true, transformation(origin = {30, 0}, extent = {{-10, 10}, {10, -10}}, rotation = 0)));
+  Modelica.Blocks.Math.Gain gain(k = Kg) annotation(
+    Placement(visible = true, transformation(origin = {150, 100}, extent = {{10, -10}, {-10, 10}}, rotation = 0)));
+
+  //Init parameters
+  parameter Types.VoltageModulePu Efd0Pu "Initial voltage output in pu (base UNom)";
+  parameter Types.CurrentModulePu Ifd0Pu "Initial current in pu (base UNom)";
```

Comment:
Two pu bases are needed for a current. Here :

(base SNom, user-selected base voltage)